### PR TITLE
feat(repo): running the sdk tests on main build

### PR DESCRIPTION
**Pull request by tsuf239**
Fixes future failures in the SDK spec test monitor.

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
     types:
       - opened
       - synchronize
+      - auto_merge_enabled
```
Comment (MarkMcCulloh) [inline .github/workflows/build.yml:8]: We don't use GH's auto merge, https://mergify.com/ handles that

Comment (tsuf239) [reply .github/workflows/build.yml]: removed then, I'll find another way in the future :) 
Can you re-review please?

File: .github/workflows/tf-aws-test.yml [hunk 2]
```diff
@@ -1,12 +1,15 @@
-name: SDK Spec AWS Tests
+name: SDK Spec AWS Tests #test
 on:
+  workflow_call: {}
+  workflow_dispatch: {}
   release:
     types:
       - published
```
Comment (MarkMcCulloh) [inline .github/workflows/tf-aws-test.yml:7]: no need for this trigger anymore

File: .github/workflows/tf-aws-test.yml [hunk 3]
```diff
@@ -68,8 +95,16 @@ jobs:
...
-          timeout_minutes: 20
-          command: wing test -t tf-aws ${{ matrix.test.directory }}/*.w
+          timeout_minutes: 30
+          command: |
+            if  ${{ env.LOCAL_BUILD == 'false' }}
+            then
+              wing test -t tf-aws ${{ matrix.test.directory }}/*.w
+            else
+              cd temp
+              ./node_modules/.bin/wing test -t tf-aws ../${{ matrix.test.directory }}/*.w
```
Comment (MarkMcCulloh) [inline .github/workflows/tf-aws-test.yml:105]:
```suggestion
              ./temp/node_modules/.bin/wing test -t tf-aws ${{ matrix.test.directory }}/*.w
```

File: .github/workflows/tf-aws-test.yml [hunk 4]
```diff
@@ -39,21 +42,45 @@ jobs:
...
+        if: ${{ env.LOCAL_BUILD == 'false' }}
         uses: nick-fields/retry@v2
         with:
           max_attempts: 3
           retry_on: error
           timeout_minutes: 5
           command: npm install -g winglang
+
       - name: Installing external js modules
+        if: ${{ env.LOCAL_BUILD == 'false' }}
```
Comment (MarkMcCulloh) [inline .github/workflows/tf-aws-test.yml:79]: This needs to happen regardless of LOCAL_BUILD, right? We aren't installing the required deps in any other step.

File: .github/workflows/tf-aws-test.yml [hunk 5]
```diff
@@ -39,21 +42,45 @@ jobs:
...
+        uses: actions/download-artifact@v3
+        with:
+          name: dist
+          path: dist
+
+      - name: Install Wing from Dist
+        if: ${{ env.LOCAL_BUILD ==  'true' }}
+        run: |
+          mkdir temp
+          cd temp
```
Comment (MarkMcCulloh) [inline .github/workflows/tf-aws-test.yml:63]: Maybe make it more descriptive

```suggestion
          mkdir localwing
          cd localwing
```

File: .github/workflows/tf-aws-test.yml [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
-  release:
-    types:
-      - published
+  workflow_call: {}
   workflow_dispatch: {}
 
 env:
   AWS_REGION: "us-east-1"
+  NODE_VERSION: "18.16.0"
+  LOCAL_BUILD: ${{ github.workflow == 'build'}}
```
[the remark this answers is not in the document]
Comment (MarkMcCulloh) [reply .github/workflows/tf-aws-test.yml:11]: > if possible

I'm fine with that as long as you've verified that `github.workflow` is actually `'build'` in a workflow_call. Looking at the latest runs it looks like you have, so 👍

**Review by MarkMcCulloh — APPROVED** [review]
Looks great, thanks for working so hard on this 🎉

**Pull request merged**